### ISSUE TITLE
[New build - DEFAULT] 24w05a: Crystals

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5459]
This build adds a new resources: crystals.
For now there is a single crystal type, used as a medium tier resource to craft lightcruises engine.
During survival the player will have to build a spaceship in order to reach asteroids where they will spawn.

Several more crystals will be added over time as higher tier resources.

What should they be called?

Suggestions
- Different item replacement logic when picking up items

Bugfixes:
#5440 Hull point increase when deleting block
#5457 Cannot exit kiosk window with Escape key
Electrolyser cannot push both hydrogen and oxygen in some cases
#5451 Ship description do not save
#5328 Fuel link can be incorrect in some position
#5454 Spaceship max speed incorrect
#5414 FPS camera wobble
#5458 Laser VFX issue
- Life support not refreshing when door closes

**Comment by Crimson-Artist** [comment]
Are asteroids going to spawn more frequently? right now asteroids only spawn in rings and not every planet has rings. Plus the distances between a planet's surface and its asteroid ring is actually extremely large. It would take forever to fly at sublight speeds just to find an asteroid to get the materials for lightcruise engines.

It would actually be better to have the lightcruise crystals to spawn planetside and have another set of crystals used in warp drives spawn on asteroids.

**Comment by Crimson-Artist** [comment]
I just ran a test. I was on a 100,000+km diameter planet with a ring. It took me about 7mins of flying at 500+m/s to go from the planet's surface to the nearest asteroid in the ring. mind you the ship I used was impractically designed to have to an extremely massive rear thruster in order to achieve that 500+m/s speed. Those 7 mins was just spent flying straight. it was very boring. Not very engaging.

My suggestion still stands. either make the lightcruise crystals spawn planetside to save players this boring flight or maybe have asteroids rain down onto planets randomly like a meteor shower. That way player don't have to do this extremely long flight and can just look for meteors in order to get their first crystals.

**Comment by CoasterBlock** [comment]
@Crimson-Artist That and boosting of sublight speed would also be nice. LCs can get you going at multiple Km or Mm per second. So I think a soft cap of around 1500-2000m per second for regular thrusters. Having "simple" crystals spawn on a planets surface  but others more advanced once needed to be found in space makes more sense too. Something about the "space exposure" would make the space crystals better. 

@tsunamayo Do you want more scifi names or something "regular"? I was thinking simple "low tier" crystals could be references to real-life materials. Such as Quartz or variants of Aluminium crystals. (Emeralds, Rubies, Saphires.)

My suggestion, based on real elements and things from Physics. 
**Tier 1:** Quartz, Alumite, Carbonite. (Silicon, Aluminium, Carbon)

**Tier 2:** Hydronium, Yttricite, Kor. (Hydrogen Metal, Yttrium, Pottasium.) 

**Tier 3:** Xenium, Neutronite, Promethyx. (Xenon, Neutrons, Prometheum.)

**Tier 4:** Gravitorium, Resonon, Zero, Queo, (Gravity, Resonance and Zero-point energy, Quantum Entanglement)

**Tier 5:** ZXN,  (can only be crafted by using all other minerals. Cannot be found. It would be the game's "Unobtanium/miracle material." )

**Comment by Janssendel** [comment]
As for the crystals, we need something that's easy to remember and unique, phonetically speaking. Crystals of a lower tier should sound more "normal" compared to the bizarre names of the highest tiered crystals. You can only use a letter *once* if they are at the front of the word to avoid confusion. 

Low Quality: Elkite, Trinitite, Algolite, Omberlux
Standard Quality: Netenkite, Snyral, Rholite, Hromium
High Quality: Leprochite, Dulcinium, Corralux, Uvinium
Exceptional Quality: Morphite, Fordinium, Pneumatite, Gildinium, Zitenkite, Quoralux, Btenki

Why 4 tiers? Crystals of Exceptional quality are rare to come by but can be fed to reactors to gain a buff to certain stats: 
-Morphite allows some regeneration of the ship's hull and an increase in ship hull points. 

-Gildinium 
allows the shields of the ship to better scatter attacks, also making the shield yellow/golden. 

-Btenki allows the propulsion systems of the ship to use less fuel and warp further. 

-Zitenkite gives all weapons a significant buff in damage (with added shake when projectiles hit)

(E) Grade crystals can be used as a substitute to regular fuel that can be used for a long time before running out, but their scarcity makes them a resource viable only during the most desperate fights.

**Comment by Janssendel** [comment]
why the strange names? that's because i based it off the most common letters in the english language

**Comment by tsunamayo** [comment]
> Are asteroids going to spawn more frequently? right now asteroids only spawn in rings and not every planet has rings. Plus the distances between a planet's surface and its asteroid ring is actually extremely large. It would take forever to fly at sublight speeds just to find an asteroid to get the materials for lightcruise engines.
> 
> It would actually be better to have the lightcruise crystals to spawn planetside and have another set of crystals used in warp drives spawn on asteroids.

Yeah I know I am still not sure about this. I have already made the ring much closer to the planet on the starting planet. I will still do some calibrating test, you will need to travel quite a bit in this game anyway, so a few minute is still acceptable. How much time does it take in space engineer to reach space from the surface? Do they have some kind of lightcruise?

**Comment by Uncle-Ulty** [comment]
I believe the crystal should receive a unique name that makes it special. I'd avoid a generic name or a known name. It must be a name that would be related to the Starship Evo itself.

Why am I saying this? because the lightdrive travel is prbly the most important and most recurrent feature that players gonna use. Something as important and special such as a lightdrive cruise, that's only possible with a lightdrive engine, must receive an important name.

-e.g. 

-evonium, evoryx, evotite, evocite, **EVOLUX**, evolium, ... you get the idea.

**Comment by tsunamayo** [comment]
@Crimson-Artist Okay so it took me 5 min with my starter ship, at 280 m/s, which is not that bad. My starting point was very far from the ring, so I will make sure to generate starting point closer to the ring. Also the starting planet is big at 120km, so I will also reduce that a bit, and greed again a bit on the ring distance to ground, and I think it should be quite acceptable.

**Comment by tsunamayo** [comment]
I like your suggestions, maybe I will open a thread so it easier to spot for everybody.

**Comment by tsunamayo** [comment]
@Crimson-Artist I did what said previously and it takes 2 min now.

**Comment by Aranhanauta** [comment]
Will crystals also be found in the rings of gas giant moons? Or just on planets?

As others have commented before, will the lower level crystals be able to stay on the surface of the planet?

**Comment by Crimson-Artist** [comment]
@tsunamayo thank you! Also do you think in the future you could add meteor showers that rain down onto planets as semi ransom events? would be nice to explore planets and suddenly asteroids crash nearby. These asteroids could be solid resource chunks such as a whole asteroid comprised of iron.
